dbt/adapters/trino/connections.py

```diff
@@ -21,32 +22,201 @@
+        if "method" in profile:
+            method = profile["method"]
+            if method == "ldap":
+                return TrinoLdapCredentials
+            elif method == "certificate":
+                return TrinoCertificateCredentials
+            elif method == "kerberos":
+                return TrinoKerberosCredentials
+            elif method == "jwt":
+                return TrinoJwtCredentials
+            elif method == "oauth":
+                return TrinoOauthCredentials
+        return TrinoNoneCredentials
+
+    @classmethod
+    def translate_aliases(
+        cls, kwargs: Dict[str, Any], recurse: bool = False
+    ) -> Dict[str, Any]:
+        klazz = cls._create_trino_profile(kwargs)
+        return klazz.translate_aliases(kwargs, recurse)
+
+    @classmethod
+    def validate(cls, data: Any):
+        klazz = cls._create_trino_profile(data)
+        return klazz.validate(data)
+
+    @classmethod
+    def from_dict(cls, data: Any):
+        klazz = cls._create_trino_profile(data)
+        return klazz.from_dict(data)
+
+
+class HttpScheme(Enum):
+    HTTP = "http"
+    HTTPS = "https"
+
+
 @dataclass
-class TrinoCredentials(Credentials):
+class TrinoNoneCredentials(Credentials):
     host: str
     port: Port
     user: str
-    password: Optional[str] = None
-    jwt_token: Optional[str] = None
-    client_certificate: Optional[str] = None
-    client_private_key: Optional[str] = None
+    cert: Optional[str] = None
     method: Optional[str] = None
+    http_scheme: HttpScheme = HttpScheme.HTTP
+    http_headers: Optional[Dict[str, str]] = None
+    session_properties: Optional[Dict[str, Any]] = None
+    _ALIASES = {"catalog": "database"}
+
+    @property
+    def type(self):
+        return "trino"
+
+    @property
+    def unique_field(self):
+        return self.host
+
+    def _connection_keys(self):
+        return ("host", "port", "user", "database", "schema", "cert")
+
+
+@dataclass
+class TrinoCertificateCredentials(Credentials):
+    host: str
+    port: Port
+    client_certificate: str
+    client_private_key: str
+    cert: Optional[str] = None
+    method: str = "certificate"
+    http_headers: Optional[Dict[str, str]] = None
+    session_properties: Optional[Dict[str, Any]] = None
+    _ALIASES = {"catalog": "database"}
+
+    @property
+    def type(self):
+        return "trino"
+
+    @property
+    def http_scheme(self):
+        return HttpScheme.HTTPS
+
+    @property
+    def unique_field(self):
+        return self.host
+
+    def _connection_keys(self):
+        return ("host", "port", "user", "database", "schema", "cert")
+
+
+@dataclass
+class TrinoLdapCredentials(Credentials):
+    host: str
+    port: Port
+    user: str
+    password: str
+    cert: Optional[str] = None
+    http_headers: Optional[Dict[str, str]] = None
+    session_properties: Optional[Dict[str, Any]] = None
+    _ALIASES = {"catalog": "database"}
+
+    @property
+    def type(self):
+        return "trino"
```

Comment:
This and `unique_field` is repeated in each Credentials class, is it feasible to have `TrinoCredentials` class and inherit from it in specific Credentials methods?

Reply:
Will try, remember hitting some limitations with inheritance and `@dataclass` classes in the past.

Reply:
Ok, that works, I think the only limitation is about the order of fields with a default value vs without default value but doesn't apply here, as long we don't try to put the properties in the base class.